app/src/main/java/org/schabi/newpipe/MainActivity.java

```diff
@@ -159,11 +160,13 @@ protected void onCreate(final Bundle savedInstanceState) {
         } catch (final Exception e) {
             ErrorActivity.reportUiErrorInSnackbar(this, "Setting up drawer", e);
         }
-
         if (DeviceUtils.isTv(this)) {
             FocusOverlayView.setupFocusObserver(this);
         }
         openMiniPlayerUponPlayerStarted();
+
+        // schedule worker for checking for new streams and creating corresponding notifications
+        NotificationWorker.schedule(this);
```

Comment:
Shouldn't it be scheduled only if notifications are enabled?

Reply:
I just realized that we never cancel the workers. Did I miss something in `NotificationsSettingsFragment` or somewhere else? I cannot find a listener for the `enable_streams_notifications` key. As a result, once a worker is running,  it is never stopped. @Stypox Can you verify that I did not miss anything, please?

If my findings are correct, I'll add a `cancel(Context)` method to the `NotificationWorker` as well as `scheduleIfNotificationsEnabled(Context) and make the required changes in the `NotificationsSettingsFragment`.

Reply:
I just pushed my proposed changes, but also ensured that the scheduled tasks are cancelled if the notifications are disabled in the system settings.

Reply:
Your solution looks good I think